packages/main/src/Input.js

```diff
@@ -670,6 +670,8 @@ class Input extends UI5Element {
 		}
 
 		this.effectiveShowClearIcon = (this.showClearIcon && !!this.value && !this.readonly && !this.disabled);
+		const numIcons = this.effectiveShowClearIcon + (this.icon ? this.icon.length : 0);
+		this.style.setProperty("--_ui5-input-icons-count", numIcons || 0);
```

Comment:
Isn't the logical operator redundant, as you are already performing the check while declaring numIcon ?